packages/realm/src/Configuration.ts

```diff
@@ -197,155 +193,73 @@ type BaseConfiguration = {
 //   disableFormatUpgrade?: boolean;
 // };
 
-// Need to use `CanonicalObjectSchema` rather than `ObjectSchema` due to some
-// integration tests using `openRealmHook()`. That function sets `this.realm`
-// to the opened realm whose schema is a `CanonicalObjectSchema[]`. Consequently,
-// the key `"ctor"` (which doesn't exist on `ObjectSchema`) also needs to be allowed.
-const OBJECT_SCHEMA_KEYS = new Set<keyof CanonicalObjectSchema>([
-  "name",
-  "primaryKey",
-  "embedded",
-  "asymmetric",
-  "properties",
-  // Not part of `ObjectSchema`
-  "ctor",
-]);
-
-// Need to use `CanonicalObjectSchemaProperty` rather than `ObjectSchemaProperty`
-// due to the same reasons as above.
-const PROPERTY_SCHEMA_KEYS = new Set<keyof CanonicalObjectSchemaProperty>([
-  "type",
-  "objectType",
-  "property",
-  "default",
-  "optional",
-  "indexed",
-  "mapTo",
-  // Not part of `ObjectSchemaProperty`
-  "name",
-]);
-
 /**
  * Validate the fields of a user-provided Realm configuration.
  */
 export function validateConfiguration(config: unknown): asserts config is Configuration {
-  assert.object(config);
-  const { path, schema, onMigration, sync } = config;
-  if (typeof onMigration !== "undefined") {
-    assert.function(onMigration, "migration");
+  assert.object(config, "realm configuration", { allowArrays: false });
+  const {
+    path,
+    schema,
+    schemaVersion,
+    inMemory,
+    readOnly,
+    fifoFilesFallbackPath,
+    sync,
+    openSyncedRealmLocally,
+    shouldCompact,
+    deleteRealmIfMigrationNeeded,
+    disableFormatUpgrade,
+    encryptionKey,
+    onMigration,
+  } = config;
+
+  if (path !== undefined) {
+    assert.string(path, "'path' on realm configuration");
+    assert(path.length > 0, "The path cannot be empty. Provide a path or remove the field.");
   }
-  if (typeof path === "string") {
-    assert(path.length > 0, "Expected a non-empty path or none at all");
+  if (schema !== undefined) {
+    validateRealmSchema(schema);
   }
-  if (onMigration && sync) {
-    throw new Error("Options 'onMigration' and 'sync' are mutually exclusive");
+  if (schemaVersion !== undefined) {
+    assert.number(schemaVersion, "'schemaVersion' on realm configuration");
```

Comment:
Should we validate the number here as well? E.g. only non-negatives (if this is the case)?